### PR TITLE
fix(settings): improve bot name validation

### PR DESCRIPTION
**Pull request by laurentlp**
This PR improves the bot name field validation by making sure that we only allow alphanumerical characters with the exceptions of some special chars (`(`, `)`, `-`, `_`, and ` `).

### CODE REVIEW

File: packages/studio-be/src/common/validation.ts [hunk 1]
```diff
@@ -4,21 +4,55 @@ export const BOTID_REGEX = /^[A-Z0-9]+[A-Z0-9_-]{1,}[A-Z0-9]+$/i
...
 export const BotEditSchema = Joi.object().keys({
-  name: Joi.string().allow('').max(50),
+  name: Joi.string()
+    .regex(/^[a-z0-9\(\)_-\s]+$/i)
+    .allow('')
+    .max(50)
+    .error(
+      () =>
+        'Bot name must not contain more than 50 alphanumerical characters. Only those special characters are accepted: (, ), -, _, and spaces.'
+    ),
```
Comment (laurentlp) [inline packages/studio-be/src/common/validation.ts:14]: Those are the only relevant changes to this file, the others being formatting.

File: packages/studio-be/src/core/app/server.ts [hunk 2]
```diff
@@ -145,15 +145,16 @@ export class HTTPServer {
...
+    return {
+      SEND_USAGE_STATS: config!.sendUsageStats,
+      USE_JWT_COOKIES: process.USE_JWT_COOKIES,
+      EXPERIMENTAL: config.experimental,
+      SOCKET_TRANSPORTS: [getSocketTransports(config).join('","')],
+      SHOW_POWERED_BY: !!config.showPoweredBy,
+      UUID: this.machineId,
+      BP_SERVER_URL: process.env.BP_SERVER_URL || '',
+      IS_STANDALONE: process.IS_STANDALONE
+    }
```
Comment (laurentlp) [inline packages/studio-be/src/core/app/server.ts:157]: I converted the JS function to an object as it simply makes more sense to transfer it via API than rendering it in the front-end.

File: packages/studio-ui/src/web/index.jsx [hunk 3]
```diff
@@ -36,35 +36,41 @@ import 'bootstrap/dist/css/bootstrap.css'
...
 
-const token = auth.getToken()
-if (token) {
-  if (window.USE_JWT_COOKIES) {
-    axios.defaults.headers.common[CSRF_TOKEN_HEADER] = token
-  } else {
-    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`
+axios.get(`${window.location.pathname || ''}/env.js`).then(({ data }) => {
+  for (const [key, value] of Object.entries(data)) {
+    window[key] = value
```
Comment (laurentlp) [inline packages/studio-ui/src/web/index.jsx:41]: Window variables are initialized here.

File: packages/studio-be/src/studio/studio-router.ts [hunk 4]
```diff
@@ -214,6 +212,31 @@ export class StudioRouter extends CustomRouter {
         res.send(totalEnv)
       })
     )
+
+    this.router.get(
+      '/labeling.js',
```
Comment (EFF) [inline packages/studio-be/src/studio/studio-router.ts:217]: being picky right now I know but can you rename this for `branding` for consistency and clarity

Comment (laurentlp) [reply packages/studio-be/src/studio/studio-router.ts:217]: Yeah good idea!

**Review by EFF — APPROVED** [review]

**Pull request merged**